UKHomeOffice/nmsw-ui
# NMSW-463 Add multiple files for upload and display list of them on page

# Ticket

NMSW-463

----

## AC

Given I have clicked on the choose files link
When I open multiple files
Then they are displayed in a 'Files added' section on the page

Given I have dragged multiple files from my file browser to the dropzone
Then they are displayed in a 'Files added' section on the page

Given I have previously added files in this session
When I add more files
Then they are added to the 'Files added' section

Given I have previously added files in this section
When I attempt to add a file with the same name as one that is already added
Then I am shown an error message

----

## To test

- create a voyage
- go to supporting documents page
- click on the choose files link
- select multiple files and open
- > you should see these files marked as Pending in the file list
- open your file browser
- drag some new files to the dropzone
- > you should see these files added as Pending in the file list
- > your original files should still be there
- click on choose files link
- try to add a file that you have already uploaded
- > you should see the browser does not add the file again 
- > you most likely won't see an error message about this file (most browsers just do not let you add a file that you've already added to it's file upload store)
- try to drag a file that you have already uploaded into the drop zone
- > you should see an error that you have already added this file 

----

## Notes

Part 1 of multiple tickets relating to file upload
